src/backends/ucsi_debugfs.rs

```diff
@@ -0,0 +1,307 @@
+                break;
+            }
+
+            let mut bitreader = BitReader::new(Cursor::new(&response[..]));
+            alternate_modes.push(AlternateMode::from_bytes(&mut bitreader)?);
+            offset += 1;
+        }
+
+        Ok(alternate_modes)
+    }
+
+    fn cable_properties(&mut self, connector_nr: usize) -> Result<CableProperty> {
+        let cmd = Command::GetCableProperty { connector_nr };
+        let response = self.execute(cmd)?;
+        let mut bitreader = BitReader::new(Cursor::new(&response[..]));
+        CableProperty::from_bytes(&mut bitreader)
+    }
+
+    fn connector_status(&mut self, _: usize) -> Result<ConnectorStatus> {
+        Err(Error::NotSupported {
+            #[cfg(feature = "backtrace")]
+            backtrace: std::backtrace::Backtrace::capture(),
+        })
+    }
+
+    fn pd_message(
+        &mut self,
+        _: usize,
+        _: MessageRecipient,
+        _: MessageResponseType,
+    ) -> Result<Message> {
+        Err(Error::NotSupported {
+            #[cfg(feature = "backtrace")]
+            backtrace: std::backtrace::Backtrace::capture(),
+        })
+    }
+
+    fn pdos(
+        &mut self,
+        connector_nr: usize,
+        partner_pdo: bool,
+        pdo_offset: u32,
+        nr_pdos: usize,
+        pdo_type: PdoType,
+        source_capabilities_type: PdoSourceCapabilitiesType,
+        revision: BcdWrapper,
+    ) -> Result<Vec<crate::pd::Pdo>> {
+        let mut pdos = vec![];
+        let mut nr_pdos_returned = 0;
+        loop {
+            if nr_pdos > 0 && nr_pdos_returned == nr_pdos {
+                break;
+            }
+
+            let cmd = Command::GetPdos {
+                connector_nr,
+                partner_pdo,
+                pdo_offset,
+                nr_pdos,
+                pdo_type,
+                source_capabilities_type,
+            };
+
+            let response = self.execute(cmd)?;
+            if response.is_null() {
+                break;
+            }
+
+            let mut bitreader = BitReader::new(Cursor::new(&response[..]));
+            let pdo = Pdo::from_bytes(&mut bitreader, revision)?;
+            pdos.push(pdo);
+
+            nr_pdos_returned += 1;
+        }
+
+        Ok(pdos)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    impl From<Driver> for UcsiDebugfsBackend {
+        fn from(mock: Driver) -> Self {
+            Self { driver: mock }
+        }
+    }
+
+    #[test]
+    fn test_stringify_command_val() {
+        let val = 12345u64;
+        let result = UcsiDebugfsBackend::stringify_command_val(val).unwrap();
+        let expected = format!("{}\0", val).into_bytes();
+
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_build_command_value_get_connector_capability() {
+        let command = Command::GetConnectorCapability { connector_nr: 0 };
+        let result = UcsiDebugfsBackend::build_command_value(&command).unwrap();
+        let expected = 65543u64;
```

Comment:
It's probably more sensible to write expected in hex (0x10007), considering the ucsi.rs specifies Command::GetConnectorCapability as 0x07.